lib/fb/maybe.ex

```diff
@@ -0,0 +1,87 @@
+defmodule FB.Maybe do
+  @moduledoc """
+  `FB.Maybe` is an explicit data type for representing values that might or might not exist.
+  """
+  @type t(a) :: {:just, a} | :nothing
+
+  @doc """
+  Creates a `FB.Maybe` representing a non-value.
+  """
+  @spec nothing() :: t(any)
+  def nothing, do: :nothing
+
+  @doc """
+  Creates a `FB.Maybe` representing value passed as an argument.
+  """
+  @spec just(a) :: t(a) when a: var
+  def just(value), do: {:just, value}
+
+  @doc """
+  Creates a `FB.Maybe` from any Elixir term.
+
+  It creates a non-value from `nil` and a value from any other term.
+
+  ## Examples
+      iex> FB.Maybe.new(nil)
+      FB.Maybe.nothing()
+
+      iex> FB.Maybe.new(:x)
+      FB.Maybe.just(:x)
+  """
+  @spec new(a | nil) :: t(a) when a: var
+  def new(term)
+  def new(nil), do: nothing()
+  def new(value), do: just(value)
+
+  @doc """
+  Transforms a `FB.Maybe` value using a provided function.
+
+  ## Examples
+      iex> FB.Maybe.map(FB.Maybe.nothing(), &String.length/1)
+      FB.Maybe.nothing()
+
+      iex> FB.Maybe.map(FB.Maybe.just("foo"), &String.length/1)
+      FB.Maybe.just(3)
+  """
+  @spec map(t(a), (a -> a)) :: t(a) when a: var
+  def map(maybe, f)
+  def map(:nothing, _), do: nothing()
+  def map({:just, value}, f), do: just(f.(value))
+
+  @doc """
+  Returns the value stored in a `FB.Maybe` or a provided default if it's a non-value. 
+
+  ## Examples
+      iex> FB.Maybe.with_default(FB.Maybe.nothing(), :default)
+      :default
+
+      iex> FB.Maybe.with_default(FB.Maybe.just(:value), :default)
+      :value
+  """
+  @spec with_default(t(a), a) :: a when a: var
+  def with_default(maybe, default)
```

Comment:
I like how this kind of function is called for Rust's Option and Result types: `unwrap_or` :)

Reply:
Yes! This name makes much more sense. Thanks!